src/main/java/org/broadinstitute/hellbender/engine/spark/BroadcastJoinReadsWithVariants.java

```diff
@@ -0,0 +1,37 @@
+package org.broadinstitute.hellbender.engine.spark;
+
+import com.google.common.collect.Lists;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.broadcast.Broadcast;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+import scala.Tuple2;
+
+import java.util.List;
+
+/**
+ * Joins an RDD of GATKReads to variant data using a broadcast strategy.
+ *
+ * The variants RDD is materialized as a List then broadcast using Spark's Broadcast variable mechanism.  The reads are
+ * then mapped over and overlapping variants are added for each read.
+ */
+public class BroadcastJoinReadsWithVariants {
+    public static JavaPairRDD<GATKRead, Iterable<Variant>> join(
+            final JavaRDD<GATKRead> reads, final JavaRDD<Variant> variants) {
+        JavaSparkContext ctx = new JavaSparkContext(reads.context());
+        Broadcast<List<Variant>> variantsBroadcast = ctx.broadcast(variants.collect());
```

Comment:
Perhaps it would be better to broadcast an `IntervalsSkipList<Variant>` instead of a `List<Variant>` to reduce the cost of `mapToPair()`? Then each `mapToPair()` could just call `variantSkipList.getOverlapping(intervalForRead)`